app/models/course/assessment/answer/programming_file.rb

```diff
@@ -10,6 +10,22 @@ class Course::Assessment::Answer::ProgrammingFile < ActiveRecord::Base
   has_many :annotations, class_name: Course::Assessment::Answer::ProgrammingFileAnnotation.name,
                          dependent: :destroy, foreign_key: :file_id, inverse_of: :file
 
+  # Returns the code from start line to end line.
+  #
+  # @param [Integer] line_start 1 based line number (first line is 1).
+  # @param [Integer] line_end 1 based line number (first line is 1).
+  # @return [String|nil] the code in the lines specified. `nil` will be returned if the input is not
+  #   valid.
+  def code_at_lines(line_start, line_end)
```

Comment:
Having a `#lines` property is a good idea, but it might be better to add one more method like
`#lines_at`, which wrappers `#lines` and accept Integer/Range as parameter and handle index conversion and overflow.